calendar.php

```diff
@@ -242,316 +242,229 @@
+        $select['repeats_5_month2'][$mybb->get_input('repeats_5_month2', MyBB::INPUT_INT)] = "selected=\"selected\"";
+        $select['repeats_5_years2'] = $mybb->get_input('repeats_5_years2', MyBB::INPUT_INT);
+
+        $timezone = $mybb->get_input('timezone', MyBB::INPUT_INT);
+    } else {
+        if (!empty($mybb->input['day'])) {
+            $day = $mybb->get_input('day', MyBB::INPUT_INT);
+        } else {
+            $day = my_date("j");
+        }
+        $select['single_day'][$day] = $select['start_day'][$day] = $select['end_day'][$day] = "selected=\"selected\"";
+
+        if (!empty($mybb->input['month'])) {
+            $month = $mybb->get_input('month', MyBB::INPUT_INT);
+        } else {
+            $month = my_date("n");
+        }
+        $select['single_month'][$month] = $select['start_month'][$month] = $select['end_month'][$month] = "selected=\"selected\"";
+
+        if (!empty($mybb->input['year'])) {
+            $single_year = $start_year = $end_year = $mybb->get_input('year', MyBB::INPUT_INT);
+        } else {
+            $single_year = $start_year = $end_year = my_date("Y");
+        }
+
+        $select['start_time'] = $select['end_time'] = '';
+        $select['type_single'] = "checked=\"checked\"";
+        $select['type_ranged'] = '';
+        $select['type'] = "single";
+        $select['repeats_1_days'] = 1;
+        $select['repeats_3_weeks'] = 1;
+        $select['repeats_4_type'][1] = "checked=\"checked\"";
+        $select['repeats_4_day'] = 1;
+        $select['repeats_4_months'] = 1;
+        $select['repeats_4_occurance'][1] = "selected=\"selected\"";
+        $select['repeats_4_weekday'][0] = "selected=\"selected\"";
+        $select['repeats_4_months2'] = 1;
+        $select['repeats_5_type'][1] = "checked=\"checked\"";
+        $select['repeats_5_day'] = 1;
+        $select['repeats_5_month'][1] = "selected=\"selected\"";
+        $select['repeats_5_years'] = 1;
+        $select['repeats_5_occurance'][1] = "selected=\"selected\"";
+        $select['repeats_5_weekday'][0] = "selected=\"selected\"";
+        $select['repeats_5_month2'][1] = "selected=\"selected\"";
+        $select['repeats_5_years2'] = 1;
+        $timezone = $mybb->user['timezone'];
+    }
+
+    $years = [];
+
+    // Construct option list for years
+    for ($year = my_date('Y'); $year < (my_date('Y') + 5); ++$year) {
+        if ($year == $single_year) {
+            $select['single_year'] = "selected=\"selected\"";
+        } else {
+            $select['single_year'] = '';
+        }
+
+        if ($year == $start_year) {
+            $select['start_year'] = "selected=\"selected\"";
+        } else {
+            $select['start_year'] = '';
+        }
+
+        if ($year == $end_year) {
+            $select['end_year'] = "selected=\"selected\"";
+        } else {
+            $select['end_year'] = '';
+        }
+
+        $years[] = $year;
+    }
+
+    $timezones = build_timezone_select("timezone", $timezone);
```

Comment:
I wonder if this is worth making a Twig function - do we use the same function elsewhere (such as the User CP)?

Reply:
As far as I'm aware it's the same everywhere. So it could be a Twig function

Reply:
Alright, sounds good. Not sure whether we should have a `TimeDateExtension`  for this and `my_date` (and whether we have any other time/date related functions to add to it) or if we should just create a `CoreExtension`/`MybbExtension` with these kinds of functions...